### PR TITLE
feat(auth): add Yahoo as provider for iOS, Android and Web

### PR DESCRIPTION
**Pull request by Lyokone**
## Description

See title

## Related Issues

*Replace this paragraph with a list of issues related to this PR from the [issue database](https://github.com/flutter/flutter/issues). Indicate, which of these issues are resolved or fixed by this PR. Note that you'll have to prefix the issue numbers with flutter/flutter#.*

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [ ] I read and followed the [Flutter Style Guide].
- [ ] I signed the [CLA].
- [ ] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [X] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/firebase/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: docs/auth/federated-auth.md [hunk 1]
```diff
@@ -343,3 +343,39 @@ with the Client ID and Secret are set, with the callback URL set in the GitHub a
...
+## Yahoo
+
+Ensure the "Yahoo" sign-in provider is enabled on the [Firebase Console](https://console.firebase.google.com/project/_/authentication/providers)
+with an API Key and API Secret set. Also make sure your Firebase OAuth redirect URI (e.g. my-app-12345.firebaseapp.com/__/auth/handler) 
+is set as your Authorization callback URL in your app's settings page on your Yahoo app's config.
+
+
+* {iOS+}
+
+  Before you begin [configure Yahoo Login for iOS](/docs/auth/ios/yahoo-oauth#before_you_begin) and add the [custom URL schemes
```
Comment (russellwheatley) [inline docs/auth/federated-auth.md:358]:
```suggestion
  Before you begin, [configure Yahoo Login for iOS](/docs/auth/ios/yahoo-oauth#before_you_begin) and add the [custom URL schemes
```

File: docs/auth/federated-auth.md [hunk 2]
```diff
@@ -343,3 +343,39 @@ with the Client ID and Secret are set, with the callback URL set in the GitHub a
...
   }
   ```
+
+
+
+## Yahoo
+
+Ensure the "Yahoo" sign-in provider is enabled on the [Firebase Console](https://console.firebase.google.com/project/_/authentication/providers)
+with an API Key and API Secret set. Also make sure your Firebase OAuth redirect URI (e.g. my-app-12345.firebaseapp.com/__/auth/handler) 
+is set as your Authorization callback URL in your app's settings page on your Yahoo app's config.
```
Comment (russellwheatley) [inline docs/auth/federated-auth.md:353]:
```suggestion
is set as your Authorization callback URL in your app's settings page on your  app's Yahoo config.
```

File: docs/auth/federated-auth.md [hunk 3]
```diff
@@ -343,3 +343,39 @@ with the Client ID and Secret are set, with the callback URL set in the GitHub a
...
+
+
+* {iOS+}
+
+  Before you begin [configure Yahoo Login for iOS](/docs/auth/ios/yahoo-oauth#before_you_begin) and add the [custom URL schemes
+  to your Runner (step 1)](https://firebase.google.com/docs/auth/ios/yahoo-oauth#handle_the_sign-in_flow_with_the_firebase_sdk).
+
+* {Android}
+  Before you begin [configure Yahoo Login for Android](/docs/auth/android/yahoo-oauth#before_you_begin).
```
Comment (russellwheatley) [inline docs/auth/federated-auth.md:362]:
```suggestion
  Before you begin, [configure Yahoo Login for Android](/docs/auth/android/yahoo-oauth#before_you_begin).
```

**Review by russellwheatley — CHANGES_REQUESTED** [review]

**Comment by russellwheatley** [comment]
Hey @kevinthecheung, do you mind reviewing the small docs changes, please?

File: docs/auth/federated-auth.md [hunk 4]
```diff
@@ -350,16 +350,17 @@ with the Client ID and Secret are set, with the callback URL set in the GitHub a
 
 Ensure the "Yahoo" sign-in provider is enabled on the [Firebase Console](https://console.firebase.google.com/project/_/authentication/providers)
 with an API Key and API Secret set. Also make sure your Firebase OAuth redirect URI (e.g. my-app-12345.firebaseapp.com/__/auth/handler) 
-is set as your Authorization callback URL in your app's settings page on your Yahoo app's config.
+is set as your Authorization callback URL in your app's settings page on your app's Yahoo config.
+.
```
Comment (kevinthecheung) [inline docs/auth/federated-auth.md:354]: Stray period

File: docs/auth/federated-auth.md [hunk 5]
```diff
@@ -343,3 +343,40 @@ with the Client ID and Secret are set, with the callback URL set in the GitHub a
...
   }
   ```
+
+
+
+## Yahoo
+
+Ensure the "Yahoo" sign-in provider is enabled on the [Firebase Console](https://console.firebase.google.com/project/_/authentication/providers)
+with an API Key and API Secret set. Also make sure your Firebase OAuth redirect URI (e.g. my-app-12345.firebaseapp.com/__/auth/handler) 
+is set as your Authorization callback URL in your app's settings page on your app's Yahoo config.
```
Comment (kevinthecheung) [inline docs/auth/federated-auth.md:353]:
```suggestion
is set as a redirect URI in your app's Yahoo Developer Network configuration.
```

**Review by kevinthecheung — APPROVED** [review]

**Review by russellwheatley — APPROVED** [review]

**Pull request merged**